database/db.go

```diff
@@ -217,14 +218,34 @@ func NewDbFromConfig(c *Config, logger *logging.Logger, connectorCallbacks Retry
 			addr = utils.JoinHostPort(c.Host, port)
 		}
 		db = sqlx.NewDb(sql.OpenDB(NewConnector(connector, logger, connectorCallbacks)), PostgreSQL)
+	case "sqlite":
+		var (
+			name = c.Database
+			mode = ""
+		)
+
+		if strings.Contains(c.Database, ":memory:") && c.Database != ":memory:" {
+			name = strings.Split(c.Database, ":memory:")[1]
+			mode = "mode=memory&"
+		}
+
+		addr = fmt.Sprintf("file:%s?%scache=shared", name, mode)
```

Comment:
As I was just fiddling with some other SQLite application, I realized that by default foreign key constraints are not enforced unless `_foreign_keys` is set accordingly. After fixing this on my side, I remembered this PR and would suggest adding this here as well.